### PR TITLE
Blog App | Create RSpec Files🎯

### PR DESCRIPTION
**Pull request by Rysth**
# 🏅 Features Implemented

List of things that were implemented based on the requirements:

## 📝 General requirements

- ✅ No Linters errors.
- ✅ Used GitFlow correctly.
- ✅ Created a professional `README` file.

## 📝 Project requirements

- ✅ Added RSpec as a gem to your project by using the gem `rspec-rails`.
- ✅ Created a Request spec file for all your controllers (PostsController, and UsersController).
- ✅ Checked the `status`, `template`, and `placeholder` text on each action (routes).

### Results
image

### RSpec UsersController
code

### RSpec PostsController
code

[_John Palacios_](https://www.linkedin.com/in/john-palacios-rysthcraft/)
_RysthCraft_

### CODE REVIEW

File: spec/requests/users_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    it 'return the body correctly' do
+      get '/users/1'
+      expect(response.body).not_to be_nil
+    end
+    it 'contains the correct placeholder text' do
+      get '/users/1'
+      expect(response.body).to include('Here is the User information.')
+    end
+  end
+end
```
Comment (Gambit142) [inline spec/requests/users_controller_spec.rb:33]: - [ ] When I run your tests, some of them fail. Kindly see the screenshot attached below for further information.

image

#### Please fix the tests and ensure that all of them pass. 👍 🖱️

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Rysth 👋, 

Good job so far!

## Highlights
- Setup linters correctly ✔️ 
- Use of Rspec library to write tests ✔️ 
- Wrote requests spec tests for `Users` and `Posts` ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/requests/posts_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    it 'return the body correctly' do
+      get '/users/1/posts/1'
+      expect(response.body).not_to be_nil
+    end
+    it 'contains the correct placeholder text' do
+      get '/users/1/posts/1'
+      expect(response.body).to include('Here is the Post information from an specific User.')
+    end
+  end
+end
```
[the remark this answers is not in the document]
Comment (Rysth) [reply spec/requests/posts_controller_spec.rb:33]: Thanks!

File: spec/requests/users_controller_spec.rb (hunk 1 above)
Comment (Rysth) [reply spec/requests/users_controller_spec.rb]: Done

[the remark this answers is not in the document]
Comment (Rysth) [reply spec/requests/users_controller_spec.rb]: Done!

File: spec/requests/posts_controller_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rysth) [reply spec/requests/posts_controller_spec.rb]: Done

File: spec/requests/posts_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    end
+
+    it 'return correct status' do
+      get "/users/#{@user.id}/posts"
+      expect(response.status).to eq(200)
+    end
+
+    it 'return the body correctly' do
+      get "/users/#{@user.id}/posts"
+      expect(response.body).not_to be_nil
```
Comment (FrederickMih) [inline spec/requests/posts_controller_spec.rb:16]: - [ ] Kindly check comment above. The `:show` template should be render correctly

File: spec/requests/posts_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    end
+
+    it 'return correct status' do
+      get "/users/#{@user.id}/posts/#{@post.id}"
+      expect(response.status).to eq(200)
+    end
+
+    it 'return the body correctly' do
+      get "/users/#{@user.id}/posts/#{@post.id}"
+      expect(response.body).not_to be_nil
```
Comment (FrederickMih) [inline spec/requests/posts_controller_spec.rb:38]: - [ ] Kindly check above and fix :wrench: this issue please.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @Rysth, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights :green_circle: 

- Rails migration errors have been successfully fix :wrench: :rocket: 
- All written tests passed :heavy_check_mark: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/users_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    end
+
+    it 'returns correct status' do
+      get "/users/#{@user.id}"
+      expect(response.status).to eq(200)
+    end
+
+    it 'returns the body correctly' do
+      get "/users/#{@user.id}"
+      expect(response.body).not_to be_nil
```
[the remark this answers is not in the document]
Comment (Rysth) [reply spec/requests/users_controller_spec.rb:33]: Thanks

File: spec/controllers/posts_controller_spec.rb
Comment (Felix45) [inline spec/controllers/posts_controller_spec.rb:1]: - [ Good job ] Good job Implementing the changes from the previous reviewer :+1: all the tests are passing now.

Screenshot from 2023-10-31 23-33-54

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @Rysth, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] All unit tests are passing :white_check_mark: 
- [x] App uses the `rspec` ruby testing gem  :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark:  
- [x] Implemented all the changes from the previous reviewer :white_check_mark:  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controllers/posts_controller_spec.rb
Comment (Rysth) [reply spec/controllers/posts_controller_spec.rb:1]: Thanks!

**Pull request merged**